Add new target support JHEMCUF405PRO (aka GHF405AIO-HD) (#29697)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/JHEMCUF405PRO/README.md

The JHEMCUF405PRO is an AIO flight controller produced by [JHEMCU](https://jhemcu.com/).

There's a different version of the board that doesn't have I2C pads exposed. IT's not that one.

## Features

 - MCU - STM32F405 32-bit processor running at 168 MHz
 - IMU - dual ICM42605
 - Barometer - DPS310
 - Voltage & current sensor
 - OSD - AT7456E
 - Onboard Flash: 8MB
 - 5x UARTs (1,2,3,4,6)
 - 5x PWM Outputs (4 Motor Output, 1 LED)
 - Battery input voltage: 2S-6S
 - BEC 5V/2.5A, 10V/2.0A
 - I2C: exposed pins
 - ESCs: BlheliS 40A (advertised)
 - LED strip: Supported
 - Buzzer: supported

## Pinout

![JHEMCUF405PRO Board](JHEMCUF405PRO.png "JHEMCUF405PRO")

## UART Mapping

The UARTs are marked RXn and TXn in the above pinouts. The RXn pin is the
receive pin for UARTn. The TXn pin is the transmit pin for UARTn.

In addition to pinouts, the board also has SH6P 1mm connector for digital FPV systems.
Please note that the board will not enter DFU mode if the receiver (eg. CRSF/ELRS is connected to either USART1/USART3/USART4). If you have it soldered there due to DMA requirement you need to temprarily desolder the wire on the FC's RX pad and solder it back after flashing

 - SERIAL0 -> USB

>  - SERIAL1 -> USART1 (DMA-enabled)

Reviewer: ```suggestion
 - SERIAL1 -> USART1 ( RCin,DMA-enabled)
```

>  - SERIAL2 -> USART2 (DMA-enabled only on Tx)

Reviewer: ```suggestion
 - SERIAL2 -> USART2 (MAVLink2, DMA-enabled only on Tx)
```

>  - SERIAL3 -> USART3 (DMA-enabled)

Reviewer: ```suggestion
 - SERIAL3 -> USART3 (GPS1, DMA-enabled)
```

>  - SERIAL4 -> UART4  (DMA-enabled only on Tx)

Reviewer: ```suggestion
 - SERIAL4 -> UART4  (GPS2, DMA-enabled only on Tx)
```

>  - SERIAL6 -> USART6 (DMA-enabled)

Reviewer: ```suggestion
 - SERIAL6 -> USART6 (MSP DisplayPort, DMA-enabled)
```

These statements are based on the current implementation of the dma_resolver.py script at the time of writing.

> ## I2C

Reviewer: ```suggestion
## Compass
```

The SCL/SDA pads are exposed as two tiny circles which you can connect a magnetometer to for example

## RC Input

> RC input is best configured on the RX1/TX1 (USART1_RX/USART1_TX) pins due to having full DMA capability and being mostly easy to access for desoldering to enable DFU mode

Reviewer: ```suggestion
RC input is best configured on the RX1/TX1 (USART1_RX/USART1_TX) pins due to having full DMA capability and being mostly easy to access for desoldering to enable DFU mode.

* PPM is not supported.
* FPort requires connection to T1 and :ref:`SERIAL1_OPTIONS<SERIAL1_OPTIONS>` be set to “7”.
* CRSF also requires a T1 connection, in addition to R1, and automatically provides telemetry. Set :ref:`SERIAL1_OPTIONS<SERIAL1_OPTIONS>` to “0”.
* SRXL2 requires a connecton to T1 and automatically provides telemetry. Set :ref:`SERIAL1_OPTIONS<SERIAL1_OPTIONS>` to “4”.

Any UART can be used for RC system connections in ArduPilot also, and is compatible with all protocols except PPM. See :ref:`Radio Control Systems <common-rc-systems>` for details.
```

## OSD Support

> JHEMCUF405PRO supports OSD using OSD_TYPE 1 (MAX7456 driver).

Reviewer: ```suggestion
* Use of internal MAX7456 for analog OSD/VTX is enabled by default.
* Simultaneous HD VTX OSD support is pre-configured on UART6 
```

## PWM Output

> JHEMCUF405PRO supports up to 5 PWM outputs. 4 motors and 1 LED strip or another PWM output.

Reviewer: ```suggestion
JHEMCUF405PRO supports up to 5 motor/servo outputs. 4 motors and 1 LED strip or another PWM output. Outputs are grouped and each group must use the same protocol. All outputs support PWM and DShot. 

Output 1-2 Group1
Output 3-4 Group 2
Output 5/LED Group 3
```

Channels 1-4 support bi-directional dshot

## Battery Monitoring

> The board has a built-in voltage and current sensor. The current

Reviewer: ```suggestion
The board has a built-in voltage sensor and l currrent sensor tied to the onboard ESC. The voltage sensor supports up to 6S LiPo.
```

> sensor's max Amps is not specified. The voltage sensor can handle up to 6S

Reviewer: ```suggestion
The default battery configuration is:

* :ref:`BATT_MONITOR<BATT_MONITOR>` = 4
* :ref:`BATT_VOLT_PIN<BATT_VOLT_PIN__AP_BattMonitor_Analog>` = 13
* :ref:`BATT_CURR_PIN<BATT_CURR_PIN__AP_BattMonitor_Analog>` = 12 
* :ref:`BATT_VOLT_MULT<BATT_VOLT_MULT__AP_BattMonitor_Analog>` = 11
* :ref:`BATT_AMP_PERVLT<BATT_AMP_PERVLT__AP_BattMonitor_Analog>` = 95.84
```

> LiPo batteries.

Reviewer: delete this

## Compass
JHEMCUF405PRO does not have a builtin compass, but you can attach an external compass using I2C on the SDA and SCL pads.

## Loading Firmware

> Initial firmware load can be done with DFU by plugging in USB with the

Reviewer: ```suggestion
Firmware for this board can be found `here <https://firmware.ardupilot.org>`__  in sub-folders labeled “JHEMCUF405PRO”.

Initial firmware load can be done with DFU by plugging in USB with the
```

bootloader button pressed. Then you should load the "with_bl.hex"
firmware, using your favourite DFU loading tool.

Once the initial firmware is loaded you can update the firmware using
any ArduPilot ground station software. Updates should be done with the

> *.apj firmware files.

Reviewer: ```suggestion
\*.apj firmware files.
```

# JHEMCUF405PRO (aka GHF405AIO-HD) Flight Controller

The JHEMCUF405PRO is an AIO flight controller produced by [JHEMCU](https://jhemcu.com/).

There's a different version of the board that doesn't have I2C pads exposed. IT's not that one.

## Features

 - MCU - STM32F405 32-bit processor running at 168 MHz

>  - IMU - dual ICM42605

Author: Fixed, single ICM42605
